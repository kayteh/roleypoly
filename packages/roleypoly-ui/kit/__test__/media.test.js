/* eslint-env jest */
import MediaQuery, { xs, sm, md, lg, xl } from '../media'

describe('MediaQuery', () => {
  it('outputs media queries', () => {
    const mq = MediaQuery({
      xs: 'font-size: 0.5em;',
      sm: 'font-size: 1em;',
      md: 'font-size: 1.5em;',
      lg: 'font-size: 2em;',
      xl: 'font-size: 2.5em;'
    })

    expect(mq).toMatchSnapshot()
  })
})

describe('mediaTemplateLiteral', () => {
  it('renders how we expect', () => {
    // this is a weird fixture because of how we render MediaQuery for testing
    const mq = `${xs`font-size: 0.5em;`}
${sm`font-size: 1em;`}
${md`font-size: 1.5em;`}
${lg`font-size: 2em;`}
${xl`font-size: 2.5em;`}`

    expect(mq).toMatchSnapshot()

    const mq2 = MediaQuery({//
      xs: 'font-size: 0.5em;',
      sm: 'font-size: 1em;',
      md: 'font-size: 1.5em;',
      lg: 'font-size: 2em;',
      xl: 'font-size: 2.5em;'
    })

    expect(mq).toEqual(mq2)
  })
})
