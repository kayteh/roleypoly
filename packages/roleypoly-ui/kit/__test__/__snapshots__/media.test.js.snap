// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaQuery outputs media queries 1`] = `
"@media screen and (min-width: 0px) {
font-size: 0.5em;
};
@media screen and (min-width: 544px) {
font-size: 1em;
};
@media screen and (min-width: 768px) {
font-size: 1.5em;
};
@media screen and (min-width: 1012px) {
font-size: 2em;
};
@media screen and (min-width: 1280px) {
font-size: 2.5em;
};"
`;

exports[`mediaTemplateLiteral renders how we expect 1`] = `
"@media screen and (min-width: 0px) {
font-size: 0.5em;
};
@media screen and (min-width: 544px) {
font-size: 1em;
};
@media screen and (min-width: 768px) {
font-size: 1.5em;
};
@media screen and (min-width: 1012px) {
font-size: 2em;
};
@media screen and (min-width: 1280px) {
font-size: 2.5em;
};"
`;
